Tests/SentryProfilerTests/SentryBacktraceTests.mm

```diff
@@ -230,7 +230,7 @@ - (void)testCollectsMultiThreadBacktrace
                     }
                 }
             },
-            cache);
+            &cache);
```

Comment:
also unsure of the ramifications of this change and the one on 202 (i can't group them into the same comment)

Reply:
OK, I missed that this is a test, but I guess I still am wondering how it works...

Reply:
I changed this test to make it compile, after @indragiek update on the code this was broken. 
I was hoping you or @indragiek to check if this is the intended use.

Reply:
You could either leave it the way it is here:

```
ThreadMetadataCache cache;
...
&cache (passing cache as a pointer to a stack allocated object)
```

or do this which is what @armcknight is suggesting above:

```
const auto cache = std::make_unique<ThreadMetadataCache>();
...
cache.get() (passing cache as a pointer to a heap allocated object)
```

In this case either will work because the pointer does not escape the implementation of `enumerateBacktracesForAllThreads`

Reply:
I will leave the way it is then. Thanks